maas/resource_maas_network_interface_bond.go

```diff
@@ -0,0 +1,310 @@
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Tags for the interface.",
+			},
+			"vlan": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "VLAN the interface is connected to.",
+			},
+		},
+	}
+}
+
+func resourceMaasNetworkInterfaceBondCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*client.Client)
+
+	machine, err := getMachine(client, d.Get("machine").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	p, err := findBondParentsID(client, machine.SystemID, d.Get("parents").(*schema.Set).List())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	params := getNetworkInterfaceBondParams(d, p)
+	networkInterface, err := client.NetworkInterfaces.CreateBond(machine.SystemID, params)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%v", networkInterface.ID))
+
+	return resourceMaasNetworkInterfaceBondRead(ctx, d, m)
+}
+
+func resourceMaasNetworkInterfaceBondRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*client.Client)
+
+	machine, err := getMachine(client, d.Get("machine").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	networkInterface, err := client.NetworkInterface.Get(machine.SystemID, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// TODO: open issue with maas client.  These fields are always ommited.
+	// They are defined in the NetworkInterface struct
+	// but are only within the Params field.
+	p := networkInterface.Params.(map[string]interface{})
+	// check if key exists within Params.
+	if _, ok := p["bond_downdelay"]; ok {
+		d.Set("bond_downdelay", int64(p["bond_downdelay"].(float64)))
+	}
+	if _, ok := p["bond_lacp_rate"]; ok {
+		d.Set("bond_lacp_rate", p["bond_lacp_rate"].(string))
+	}
+	if _, ok := p["bond_miimon"]; ok {
+		d.Set("bond_miimon", int64(p["bond_miimon"].(float64)))
+	}
+	if _, ok := p["bond_mode"]; ok {
+		d.Set("bond_mode", p["bond_mode"].(string))
+	}
+	if _, ok := p["bond_num_grat_arp"]; ok {
+		d.Set("bond_num_grat_arp", int64(p["bond_num_grat_arp"].(float64)))
+	}
+	if _, ok := p["bond_updelay"]; ok {
+		d.Set("bond_updelay", int64(p["bond_updelay"].(float64)))
+	}
+	if _, ok := p["bond_xmit_hash_policy"]; ok {
+		d.Set("bond_xmit_hash_policy", p["bond_xmit_hash_policy"].(string))
+	}
+
+	tfState := map[string]interface{}{
+		"name":        networkInterface.Name,
+		"parents":     networkInterface.Parents,
+		"accept_ra":   networkInterface.AcceptRA,
+		"autoconf":    networkInterface.Autoconf,
+		"mac_address": networkInterface.MACAddress,
+		"mtu":         networkInterface.EffectiveMTU,
+		"tags":        networkInterface.Tags,
+		"vlan":        fmt.Sprintf("%v", networkInterface.VLAN.ID),
```

Comment:
`networkInterface.VLAN.ID` should always be an int, can you please use `strconv.Itoa()` here?